FluentTc/Locators/GitVCSRootBuilder.cs

```diff
@@ -0,0 +1,210 @@
+                new Property()
+                {
+                    Name = name,
+                    Value = value
+                }
+            );
+        }
+
+        private void AppendProperty<T>(T value)
+        {
+            m_VCSRoot.Properties.Property.Add(
+                new Property()
+                {
+                    Name = typeof(T).Name.FromPascalToCamelCase(),
+                    Value = value.ToString().FromPascalToCapitalizedCase()
+                }
+            );
+        }
+
+        public IGitVCSRootBuilder AgentCleanFilePolicy(AgentCleanFilePolicy value)
+        {
+            AppendProperty(value);
+            return this;
+        }
+
+        public IGitVCSRootBuilder AgentCleanPolicy(AgentCleanPolicy value)
+        {
+            AppendProperty(value);
+            return this;
+        }
+
+        public IGitVCSRootBuilder AuthMethod(AuthMethod value)
+        {
+            AppendProperty(value);
+            return this;
+        }
+
+        public IGitVCSRootBuilder Branch(string value)
+        {
+            AppendProperty("branch", value);
+            return this;
+        }
+
+        public IGitVCSRootBuilder IgnoreKnownHosts()
+        {
+            AppendProperty("ignoreKnownHosts", "true");
+            return this;
+        }
+
+        public IGitVCSRootBuilder DontIgnoreKnownHosts()
+        {
+            AppendProperty("ignoreKnownHosts", "false");
+            return this;
+        }
+
+        public IGitVCSRootBuilder UserNameStyle(UserNameStyle value)
+        {
+            AppendProperty(value);
+            return this;
+        }
+
+        public IGitVCSRootBuilder Username(string value)
+        {
+            AppendProperty("username", value);
+            return this;
+        }
+
+        public IGitVCSRootBuilder Password(string value)
+        {
+            AppendProperty("secure:password", value);
+            return this;
+        }
+
+        public IGitVCSRootBuilder DontUseAlternates()
+        {
+            AppendProperty("useAlternates", "false");
+            return this;
+        }
+
+        public IGitVCSRootBuilder UseAlternates()
+        {
+            AppendProperty("useAlternates", "true");
+            return this;
+        }
+
+        public IGitVCSRootBuilder Url(Uri value)
+        {
+            AppendProperty("url", value.ToString());
+            return this;
+        }
+
+        public IGitVCSRootBuilder BranchSpec(string value)
+        {
+            AppendProperty("teamcity:branchSpec", value);
+            return this;
+        }
+
+        public IGitVCSRootBuilder CheckoutSubModule()
+        {
+            AppendProperty("submoduleCheckout", "CHECKOUT");
+            return this;
+        }
+
+        public IGitVCSRootBuilder DontCheckoutSubModule()
+        {
+            AppendProperty("submoduleCheckout", "IGNORE");
+            return this;
+        }
+
+        public IGitVCSRootBuilder Project(Project value)
```

Comment:
If we stick with the fluent API then this method should also accept lambda. There is a project locator for this purpose:
```C#
.Project(_ => _.Id("ProjectId"))
```
What do you think?